meson.build

```diff
@@ -167,25 +169,20 @@ if compiler.get_id() != 'msvc'
   override_options += 'cpp_std=c++11'
 endif
 
-if tracy_shared_libs
-  tracy = shared_library('tracy', tracy_src, tracy_header_files,
-    dependencies        : [ threads_dep ],
-    include_directories : tracy_public_include_dirs,
-    override_options    : override_options,
-    install             : true)
-else
-  tracy = static_library('tracy', tracy_src, tracy_header_files,
+tracy_compile_args += tracy_common_args
+
+tracy = library('tracy', tracy_src, tracy_header_files,
```

Comment:
As far as I can tell, if `default_library` is `static`, then this will build a static library, even if `tracy_shared_libs` is enabled. I would consider this a breaking change, as using a static Tracy capture library instead of shared can cause assertions with certain multi-DLL application setups. So, it might be better to just drop the `tracy_shared_libs` option in this case.

Reply:
Perhaps an `error` if `tracy_shared_libs=true` and `default_library=static` would be a good intermediate step?, `option` inside the `meson_options.txt` has `deprecated` since 0.60 but I opted to avoid it as to not increase the minimum required meson version. If bumping the min version to 0.60 is acceptable then that could be an option.

Reply:
But then there's the inverse problem that `default_library=shared` and `tracy_shared_libs=false` will build a shared lib, right? Kinda weird still. I guess it's a call for @wolfpld on the option dropping policy here.

Reply:
Please add the deprecation tag. The Debian people will find ways to complain that something doesn't work in their museum either way.

Reply:
added